service/frontend/workflow_handler.go

```diff
@@ -4662,3 +4686,68 @@ func getBatchOperationState(workflowState enumspb.WorkflowExecutionStatus) enums
 	}
 	return operationState
 }
+
+func (wh *WorkflowHandler) mapStartWorkflowExecutionRequestSearchAttributes(request *workflowservice.StartWorkflowExecutionRequest, namespaceName namespace.Name) (*workflowservice.StartWorkflowExecutionRequest, error) {
+	mappedSearchAttributes, err := searchattribute.SubstituteAliases(wh.saMapper, request.GetSearchAttributes(), namespaceName.String())
+	if err != nil {
+		return nil, err
+	}
+	if mappedSearchAttributes == nil {
```

Comment:
I think going out of your way to avoid a copy here is unnecessary. The shallow copies are cheap (sure, avoid the deep copy). If we have a mapper at all, we're almost certainly going to be mapping. So the simpler logic could be:
```
if wh.saMapper == nil || request.SearchAttributes == nil { return }
otherwise map and copy request
```
By the time we invoke the mapper we've already allocated a new map and filled it up, so not much point avoiding another copy